### PR TITLE
[realvision] Add extractor for RealVision.

### PR DESCRIPTION
**Pull request by jsharkey**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Real Vision examines finance, business and the global economy with
videos and reports that provide real in-depth analysis from real
experts.

Internally they're using Brightcove for content distribution, but we
need to make several requests to discover the Realvision video ID
which looks like an MD5 hash, and then use that to discover the
Brightcove video ID.  Along the way we collect better metadata like
published-vs-filmed dates, and a cleaner thumbnail.

Some videos are free, but others behind the membership wall require
the usage of a cookie jar.

### CODE REVIEW

File: youtube_dl/extractor/realvision.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from __future__ import unicode_literals
```
Comment (lukasz-kaniowski) [inline youtube_dl/extractor/realvision.py:1]:
```suggestion
# coding: utf-8
from __future__ import unicode_literals
```

**Comment by lukasz-kaniowski** [comment]
Hi @jsharkey, I've tried your branch and I'm getting following error for videos behind the paywall
```
python3 -m youtube_dl --cookies rv-cookies.txt https://www.realvision.com/shows/the-interview/videos/meet-the-long-vol-kingmaker --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--cookies', 'rv-cookies.txt', 'https://www.realvision.com/shows/the-interview/videos/meet-the-long-vol-kingmaker', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.06.16.1
[debug] Git HEAD: e9fa2a6e7
[debug] Python version 3.8.3 (CPython) - macOS-10.15.5-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3, ffprobe 4.3, rtmpdump 2.4
[debug] Proxy map: {}
[RealVision] https://www.realvision.com/shows/the-interview/videos/meet-the-long-vol-kingmaker: Downloading webpage
[RealVision] 4cd77be6d05849e8be17b7035da08651: Downloading webpage
ERROR: An extractor error has occurred. (caused by KeyError('videoassets_brightcove_id')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/extractor/realvision.py", line 43, in _real_extract
    'url': self.BRIGHTCOVE_URL_TEMPLATE % (metadata['included'][0]['attributes']['videoassets_brightcove_id']),
KeyError: 'videoassets_brightcove_id'
Traceback (most recent call last):
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/extractor/realvision.py", line 43, in _real_extract
    'url': self.BRIGHTCOVE_URL_TEMPLATE % (metadata['included'][0]['attributes']['videoassets_brightcove_id']),
KeyError: 'videoassets_brightcove_id'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/extractor/common.py", line 543, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
youtube_dl.utils.ExtractorError: An extractor error has occurred. (caused by KeyError('videoassets_brightcove_id')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

I've exported cookies for rv following readme instructions. Is there something that I'm missing?

File: youtube_dl/extractor/realvision.py (hunk 1 above)
Comment (jsharkey) [reply youtube_dl/extractor/realvision.py:2]: Done, thanks for the pointer.

**Comment by jsharkey** [comment]
Hmm, thanks for trying it with a video behind the paywall; are you able to play that video when logged into the site in a normal web browser?

I wonder if it's using a non-Brightcove source.  I just pushed an updated CL that should be generally more robust to missing metadata, and it'll provide a bit more context about where/how it failed.  PTAL?

**Comment by lukasz-kaniowski** [comment]
I'm able to play video when logged in. I can also download the video using `.m3u8` link that I can find in the network tab of chrome dev tools.

Current error
```
➜  youtube-dl git:(jsharkey/master) ✗ python3 -m youtube_dl --cookies rv-cookies.txt https://www.realvision.com/shows/the-interview/videos/meet-the-long-vol-kingmaker --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--cookies', 'rv-cookies.txt', 'https://www.realvision.com/shows/the-interview/videos/meet-the-long-vol-kingmaker', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.06.16.1
[debug] Git HEAD: c3ffe39c3
[debug] Python version 3.8.3 (CPython) - macOS-10.15.5-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3, ffprobe 4.3, rtmpdump 2.4
[debug] Proxy map: {}
[RealVision] https://www.realvision.com/shows/the-interview/videos/meet-the-long-vol-kingmaker: Downloading webpage
[RealVision] 4cd77be6d05849e8be17b7035da08651: Downloading JSON metadata
ERROR: No Brightcove assets for 4cd77be6d05849e8be17b7035da08651.
Traceback (most recent call last):
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/Users/lukasz/Projects/opensource/youtube-dl/youtube_dl/extractor/realvision.py", line 58, in _real_extract
    raise ExtractorError('No Brightcove assets for %s.' % rv_id, expected=True)
youtube_dl.utils.ExtractorError: No Brightcove assets for 4cd77be6d05849e8be17b7035da08651.
```

**Pull request closed without merging**